.github/workflows/run-unit-tests.yml

```diff
@@ -179,7 +179,8 @@ jobs:
       - name: >
           Migration Tests: ${{ matrix.python-version }}:${{ env.PARALLEL_TEST_TYPES }}
         uses: ./.github/actions/migration_tests
-        if: inputs.run-migration-tests == 'true' && inputs.test-group == 'core'
+        # TODO(potiuk) - we should bring back migration tests when we move the migration to start from Airflow 3
```

Comment:
Means: Migration fails if the migration attempts to run from a legacy version Airflow <3.0? So if you want to use python 3.13 you need to use Py 3.12 to migrate to 3.0 at least and then you can continue on top of this?

Reply:
I am not sure why it did not work  - the TODO: is here so that I can bring it back and investigate - becaus it was blocking the "regular" tests. But before I merge it, I need to figure out what is going on here.

Reply:
> Means: Migration fails if the migration attempts to run from a legacy version Airflow <3.0? So if you want to use python 3.13 you need to use Py 3.12 to migrate to 3.0 at least and then you can continue on top of this?

I had to do a little more gymnastics with that and had to complicate the migration tests:

* the first part of migration where we create the db using older airflow version to create the DB will not work - because we cannot install older versions of airflow (2.11.0) with Python 3.13
* we should not set FABDBManager at all for Python 3.13 as we have no FAB
* so we needed separate steps dedicated to Python 3.13.